pkg/monitoring/monitoring.go

```diff
@@ -3,104 +3,109 @@ package monitoring
 import (
 	"context"
 	"fmt"
+	"log"
+	"math"
+	"net"
 	"net/http"
 	"net/http/pprof"
+	"strconv"
 	"strings"
 
 	"github.com/giongto35/cloud-game/v2/pkg/config/monitoring"
-	config "github.com/giongto35/cloud-game/v2/pkg/config/worker"
-	"github.com/golang/glog"
+	"github.com/giongto35/cloud-game/v2/pkg/network/httpx"
+	"github.com/giongto35/cloud-game/v2/pkg/service"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
-type ServerMonitoring struct {
-	cfg    monitoring.ServerMonitoringConfig
+const debugEndpoint = "/debug/pprof"
+const metricsEndpoint = "/metrics"
+
+type Monitoring struct {
+	service.RunnableService
+
+	conf   monitoring.Config
 	tag    string
-	server *http.Server
+	server *httpx.Server
 }
 
-func NewServerMonitoring(cfg monitoring.ServerMonitoringConfig, tag string) *ServerMonitoring {
-	return &ServerMonitoring{cfg: validate(&cfg), tag: tag}
+// New creates new monitoring service.
+// The tag param specifies owner label for logs.
+func New(conf monitoring.Config, baseAddr string, tag string) *Monitoring {
+	serv, err := httpx.NewServer(
+		net.JoinHostPort(baseAddr, strconv.Itoa(conf.Port)),
```

Comment:
I want to mention that right now when the main http/s listener is created, its address (host:port, the port can be auto-generated [:0] or rolled [port+1]) is passed into monitoring or whatever needs it.
Here we merge the actual host value from the main listener with a monitoring port. Port can be rolled in worker/coordinator. The resulting address will be printed into the console.